File: December 2022/20 December 2022/Python/A difference of values and indexes.py
class Solution:
    def maxDistance (self, arr, n) : 
        max1=max2=-float("inf")
        min1=min2=float("inf")
        for i in range(n):
            max1=max(max1,arr[i]+(i+1))
            min1=min(min1,arr[i]+(i+1))
            max2=max(max2,arr[i]-(i+1))
            min2=min(min2,arr[i]-(i+1))
        return max(max1-min1,max2-min2)

#{ 
 # Driver Code Starts
#Initial Template for Python 3


for _ in range(0,int(input())):
    n = int(input())
    arr = list(map(int, input().strip().split()))
    ans = Solution().maxDistance(arr, n)
    print(ans)

# } Driver Code Ends

File: December 2022/20 December 2022/Java/A difference of values and indexes.java
//{ Driver Code Starts
//Initial Template for Java

//Initial Template for Java


/*package whatever //do not write package name here */

import java.io.*;
import java.util.*;


class Array {
    
    // Driver code
	public static void main (String[] args) throws IOException{
		// Taking input using buffered reader
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		int testcases = Integer.parseInt(br.readLine());
		
		// looping through all testcases
		while(testcases-- > 0){
		    String line = br.readLine();
		    String[] element = line.trim().split("\\s+");
		    int n = Integer.parseInt(element[0]);
		     
		    int arr [] = new int[n];
		    
		    line = br.readLine();
		    String[] elements = line.trim().split("\\s+");
		    for(int i = 0;i<n;i++){
		        arr[i] = Integer.parseInt(elements[i]);
		    }
		    Solution obj = new Solution();
		    int ans = obj.maxDistance(arr, n);
		    System.out.println(ans);
		}
	}
}




// } Driver Code Ends

class Solution{
    public static int maxDistance (int arr[], int n) {
        int min1=Integer.MAX_VALUE,min2=Integer.MAX_VALUE;
        int max1=Integer.MIN_VALUE,max2=Integer.MIN_VALUE;
        int ans=0;
        for(int i=0;i<n;i++){
            max1=Math.max(max1,arr[i]+(i+1));
            min1=Math.min(min1,arr[i]+(i+1));
            max2=Math.max(max2,arr[i]-(i+1));
            min2=Math.min(min2,arr[i]-(i+1));
        }
        return Math.max(max1-min1,max2-min2);
    }
}